Tools/autotest/arduplane.py

```diff
@@ -4160,6 +4160,38 @@ def LandingDrift(self):
 
         self.wait_disarmed(timeout=180)
 
+    def CatapultTakeoff(self):
+        '''Test that a catapult takeoff works correctly'''
+
+        self.set_parameters({
+            "TKOFF_THR_MINACC": 3.0,
+            "RTL_AUTOLAND": 2, # The mission contains a DO_LAND_START item.
+        })
+        # Record desired target speed.
+        min_speed = self.get_parameter("AIRSPEED_CRUISE")
+        max_throttle = self.get_parameter("TKOFF_THR_MAX")
+        if max_throttle == 0:
+            max_throttle = self.get_parameter("THR_MAX")
+
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+
+        # Load and start mission.
+        self.load_mission("catapult.txt", strict=True)
+        self.set_current_waypoint(1, check_afterwards=True)
+        self.change_mode('AUTO')
```

Comment:
Usually you'd load your mission before arming your vehicle :-)

I suggest using `wait_groundspeed(..., minimum_duration=5)` to ensure the vehicle doesn't move until the catapult is released.

Reply:
I'll place the mission loading before and just wait for arming checks to pass.
The mission load ended there in an act of frustration ;)